File: src/main/java/com/yessinmedqa/models/Question.java
package com.yessinmedqa.models;

public class Question {

    private int id;
    private int userId;
    private String categorie;
    private String titre;
    private String description;
    private boolean sousTraitement;
    private boolean aDesAllergies;
    private String fichierChemin;
    private String statut;
    private double taille;
    private double poids;
    private String descriptionTraitement;
    private String descriptionAllergies;

    public Question() {}

    public Question(int userId, String titre, String description,
                    boolean sousTraitement, boolean aDesAllergies,
                    String fichierChemin, double taille, double poids,
                    String descriptionTraitement, String descriptionAllergies) {
        this.userId = userId;
        this.titre = titre;
        this.description = description;
        this.sousTraitement = sousTraitement;
        this.aDesAllergies = aDesAllergies;
        this.fichierChemin = fichierChemin;
        this.taille = taille;
        this.poids = poids;
        this.descriptionTraitement = descriptionTraitement;
        this.descriptionAllergies = descriptionAllergies;
        this.statut = "en_attente";
    }

    // Getters and Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getUserId() { return userId; }
    public void setUserId(int userId) { this.userId = userId; }

    public String getCategorie() { return categorie; }
    public void setCategorie(String categorie) { this.categorie = categorie; }

    public String getTitre() { return titre; }
    public void setTitre(String titre) { this.titre = titre; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public boolean isSousTraitement() { return sousTraitement; }
    public void setSousTraitement(boolean sousTraitement) { this.sousTraitement = sousTraitement; }

    public boolean isaDesAllergies() { return aDesAllergies; }
    public void setaDesAllergies(boolean aDesAllergies) { this.aDesAllergies = aDesAllergies; }

    public String getFichierChemin() { return fichierChemin; }
    public void setFichierChemin(String fichierChemin) { this.fichierChemin = fichierChemin; }

    public String getStatut() { return statut; }
    public void setStatut(String statut) { this.statut = statut; }

    public double getTaille() { return taille; }
    public void setTaille(double taille) { this.taille = taille; }

    public double getPoids() { return poids; }
    public void setPoids(double poids) { this.poids = poids; }

    public String getDescriptionTraitement() { return descriptionTraitement; }
    public void setDescriptionTraitement(String descriptionTraitement) { this.descriptionTraitement = descriptionTraitement; }

    public String getDescriptionAllergies() { return descriptionAllergies; }
    public void setDescriptionAllergies(String descriptionAllergies) { this.descriptionAllergies = descriptionAllergies; }
}

File: symfony-integ/src/Entity/Question.php
<?php

namespace App\Entity;

use App\Repository\QuestionRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: QuestionRepository::class)]
#[ORM\Table(name: 'questions')]
class Question
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(name: 'user_id')]
    private ?int $userId = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $categorie = null;

    #[ORM\Column(length: 255)]
    private ?string $titre = null;

    #[ORM\Column(type: 'text')]
    private ?string $description = null;

    #[ORM\Column(name: 'sous_traitement', nullable: true)]
    private ?bool $sousTraitement = null;

    #[ORM\Column(name: 'a_des_allergies', nullable: true)]
    private ?bool $aDesAllergies = null;

    #[ORM\Column(name: 'fichier_chemin', length: 255, nullable: true)]
    private ?string $fichierChemin = null;

    #[ORM\Column(length: 50, options: ['default' => 'en_attente'])]
    private string $statut = 'en_attente';

    #[ORM\Column(nullable: true)]
    private ?float $taille = null;

    #[ORM\Column(nullable: true)]
    private ?float $poids = null;

    #[ORM\Column(name: 'description_traitement', type: 'text', nullable: true)]
    private ?string $descriptionTraitement = null;

    #[ORM\Column(name: 'description_allergies', type: 'text', nullable: true)]
    private ?string $descriptionAllergies = null;

    #[ORM\Column(name: 'created_at', type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $createdAt = null;

    // Non-mapped field for user display
    private ?string $userNom = null;
    private ?string $userPrenom = null;

    public function getId(): ?int { return $this->id; }
    public function getUserId(): ?int { return $this->userId; }
    public function setUserId(int $userId): static { $this->userId = $userId; return $this; }
    public function getCategorie(): ?string { return $this->categorie; }
    public function setCategorie(?string $categorie): static { $this->categorie = $categorie; return $this; }
    public function getTitre(): ?string { return $this->titre; }
    public function setTitre(string $titre): static { $this->titre = $titre; return $this; }
    public function getDescription(): ?string { return $this->description; }
    public function setDescription(string $description): static { $this->description = $description; return $this; }
    public function isSousTraitement(): ?bool { return $this->sousTraitement; }
    public function setSousTraitement(?bool $v): static { $this->sousTraitement = $v; return $this; }
    public function isADesAllergies(): ?bool { return $this->aDesAllergies; }
    public function setADesAllergies(?bool $v): static { $this->aDesAllergies = $v; return $this; }
    public function getFichierChemin(): ?string { return $this->fichierChemin; }
    public function setFichierChemin(?string $v): static { $this->fichierChemin = $v; return $this; }
    public function getStatut(): string { return $this->statut; }
    public function setStatut(string $statut): static { $this->statut = $statut; return $this; }
    public function getTaille(): ?float { return $this->taille; }
    public function setTaille(?float $taille): static { $this->taille = $taille; return $this; }
    public function getPoids(): ?float { return $this->poids; }
    public function setPoids(?float $poids): static { $this->poids = $poids; return $this; }
    public function getDescriptionTraitement(): ?string { return $this->descriptionTraitement; }
    public function setDescriptionTraitement(?string $v): static { $this->descriptionTraitement = $v; return $this; }
    public function getDescriptionAllergies(): ?string { return $this->descriptionAllergies; }
    public function setDescriptionAllergies(?string $v): static { $this->descriptionAllergies = $v; return $this; }
    public function getCreatedAt(): ?\DateTimeInterface { return $this->createdAt; }
    public function setCreatedAt(?\DateTimeInterface $v): static { $this->createdAt = $v; return $this; }
    public function getUserNom(): ?string { return $this->userNom; }
    public function setUserNom(?string $v): static { $this->userNom = $v; return $this; }
    public function getUserPrenom(): ?string { return $this->userPrenom; }
    public function setUserPrenom(?string $v): static { $this->userPrenom = $v; return $this; }
}
